Sources/SwiftInspectorCore/Tests/PropertyAnalzyerSpec.swift

```diff
@@ -0,0 +1,580 @@
+                        public final class FakeType {}
+                        """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("returns type info with empty property list") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties).to(beEmpty())
+        }
+      }
+
+      context("when there is a property") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            public var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the type name") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.name) == "FakeType"
+        }
+
+        it("returns nil if the type name is not present") {
+          let sut = PropertyAnalyzer(typeName: "AnotherType")
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result).to(beNil())
+        }
+
+        it("detects the properties") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeProperties.PropertyData(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "",
+              modifiers: [.public, .instance])
+          ]
+        }
+      }
+
+      context("when there is a property in a nested type with the same type name") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            public var thing: String = "Hello, World"
+
+            enum FakeType {
+              static let foo: String = "Hola"
+            }
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the type name") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.name) == "FakeType"
+        }
+
+        /*
+         This is actually not the ideal result of this and is a limitation of the implementation
+         Ideally you would have to pass in `FakeType.FakeType` to get this nested type's property
+         information. For now we are accepting this limitation and have this test to showcase
+         what happens in this scenario.
+         */
+        it("detects and merges the properties") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          let propSet = Set(result?.properties ?? [])
+          let expectedPropSet: Set<TypeProperties.PropertyData> = [
+            .init(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "",
+              modifiers: [.public, .instance]),
+            .init(
+              name: "foo",
+              typeAnnotation: "String",
+              comment: "",
+              modifiers: [.internal, .static])
+          ]
+          expect(propSet) == expectedPropSet
+        }
```

Comment:
Agreed with Michael, it's great to have a test like this and document that it's not ideal but it's what a workaround we're ok with.